dot/core/service.go

```diff
@@ -340,13 +345,29 @@ func (s *Service) handleChainReorg(prev, curr common.Hash) error {
 		for _, ext := range exts {
 			logger.Debug("validating transaction on re-org chain", "extrinsic", ext)
 
-			txv, err := s.rt.ValidateTransaction(ext)
+			decExt := &types.ExtrinsicData{}
+			err = decExt.DecodeVersion(ext)
+			if err != nil {
+				return err
+			}
+
+			// Inherent are not signed.
+			if !decExt.IsSigned() {
+				continue
+			}
+
+			encExt, err := scale.Encode(ext)
+			if err != nil {
+				return err
+			}
+
+			txv, err := s.rt.ValidateTransaction(append([]byte{byte(types.TxnExternal)}, encExt...))
```

Comment:
nit: move `append([]byte{byte(types.TxnExternal)}, encExt...)` into it's own line/var

Reply:
Done